reference-implementation/lib/resolver.js

```diff
@@ -1,100 +1,60 @@
+  const fallbacks = exports.getFallbacks(taggedSpecifier.specifier, parsedImportMap, scriptURL.href);
+  for (const address of fallbacks) {
+    const taggedSpecifierValue = tryURLLikeSpecifierParse(address, scriptURL);
+    if (taggedSpecifierValue.type !== 'url') {
+      throw new TypeError(`The specifier ${JSON.stringify(specifier)} was resolved to ` +
+        `non-URL ${JSON.stringify(address)}.`);
+    }
+    if (!taggedSpecifierValue.isBuiltin || supportedBuiltInModules.has(taggedSpecifierValue.specifier)) {
+      return new URL(taggedSpecifierValue.specifier);
     }
-    return asURL;
   }
-
-  throw new TypeError(`Unmapped bare specifier "${specifier}"`);
+  throw new TypeError(`The specifier ${JSON.stringify(specifier)} could not be resolved.`);
 };
 
-function resolveImportsMatch(normalizedSpecifier, specifierMap) {
-  for (const [specifierKey, addresses] of Object.entries(specifierMap)) {
-    // Exact-match case
-    if (specifierKey === normalizedSpecifier) {
-      if (addresses.length === 0) {
-        throw new TypeError(`Specifier "${normalizedSpecifier}" was mapped to no addresses.`);
-      } else if (addresses.length === 1) {
-        const singleAddress = addresses[0];
-        if (singleAddress.protocol === BUILT_IN_MODULE_PROTOCOL && !supportedBuiltInModules.has(singleAddress.href)) {
-          throw new TypeError(`The "${singleAddress.href}" built-in module is not implemented.`);
-        }
-        return singleAddress;
-      } else if (addresses.length === 2 &&
-                 addresses[0].protocol === BUILT_IN_MODULE_PROTOCOL &&
-                 addresses[1].protocol !== BUILT_IN_MODULE_PROTOCOL) {
-        return supportedBuiltInModules.has(addresses[0].href) ? addresses[0] : addresses[1];
-      } else {
-        throw new Error('The reference implementation for multi-address fallbacks that are not ' +
-                        '[built-in module, fetch-scheme URL] is not yet implemented.');
-      }
-    }
-
-    // Package prefix-match case
-    if (specifierKey.endsWith('/') && normalizedSpecifier.startsWith(specifierKey)) {
-      if (addresses.length === 0) {
-        throw new TypeError(`Specifier "${normalizedSpecifier}" was mapped to no addresses ` +
-                            `(via prefix specifier key "${specifierKey}").`);
-      } else if (addresses.length === 1) {
-        const afterPrefix = normalizedSpecifier.substring(specifierKey.length);
-
-        // Enforced by parsing
-        assert(addresses[0].href.endsWith('/'));
-
-        // Cannot use URL resolution directly. E.g. "switch" relative to "std:elements/" is a
-        // parsing failure.
-        return new URL(addresses[0] + afterPrefix);
-      } else if (addresses.length === 2 &&
-                 addresses[0].protocol === BUILT_IN_MODULE_PROTOCOL &&
-                 addresses[1].protocol !== BUILT_IN_MODULE_PROTOCOL) {
+exports.getFallbacks = (normalizedSpecifier, parsedImportMap, resolutionContext) => {
```

Comment:
resolutionContext -> scriptURL, maybe?

Reply:
This is used in [the composer](https://github.com/WICG/import-maps/blob/ccaf0c062738a2c1b93470ffffa297ed3f251b33/reference-implementation/lib/composer.js#L23) to resolve fallbacks for a scope in the context of the existing map based on that scope (i.e., with the scope as the resolution context). In that case there's not yet a "script"; hence the name of this parameter.